tests/FSharp.Compiler.ComponentTests/Attributes/AttributePrivateSetterTests.fs

```diff
@@ -0,0 +1,45 @@
+namespace FSharp.Compiler.ComponentTests
+
+open Xunit
+open FSharp.Test.Compiler
+
+[<TestFixture>]
+type AttributePrivateSetterTests() =
+
+    [<Fact>]
+    let ``Compiler disallows setting a value of a private property on attribute``() =
+        // Arrange
+        let code = """
```

Comment:
This is too complicated.
Please use a BCL attribute which has private setter instead - you can use  the `System.Diagnostics.ConditionalAttribute` from BCL and its `ConditionString` property - this one has a private setter.

Reply:
That way, you can have a standalone F# example and do not have to deal with C# and multiple sources.